components/script/timers.rs

```diff
@@ -231,8 +349,67 @@ impl OneshotTimers {
             if !global.can_continue_running() {
                 return;
             }
-            let callback = timer.callback;
-            callback.invoke(global, &self.js_timers, can_gc);
+            match &timer.callback {
+                OneshotTimerCallback::RunStepsAfterTimeout { ordering_id, .. } => {
```

Comment:
So this should end-up turned inside out, where the "normal" timers are just using the "run steps after timeout" mechanism and an ordering id of "setTimeout/setInterval". Let's file an issue for that. 

The current implementation means that you could actually just use `SetTimeOut`, but the way the spec works is rather that this is the general mechanism, and `SetTimeOut` is a specific use of it.